src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/BackgroundAspectEffect/BackgroundAspectEffectRouter.ios.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using UIKit;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+using Effects = Xamarin.CommunityToolkit.iOS.Effects;
+
+[assembly: ExportEffect(typeof(Effects.BackgroundAspectEffectRouter), nameof(BackgroundAspectEffectRouter))]
+
+namespace Xamarin.CommunityToolkit.iOS.Effects
+{
+	public class BackgroundAspectEffectRouter : PlatformEffect
+	{
+		UIImageView? imageView;
+
+		CancellationTokenSource cancellationTokenSource;
+
+		public BackgroundAspectEffectRouter()
+		{
+			cancellationTokenSource = new CancellationTokenSource();
+		}
+
+		protected override void OnAttached()
+			=> Task.Run(ApplyBackgroundImageAsync, cancellationTokenSource.Token);
+
+		protected override void OnDetached()
+			=> ClearBackgroundImage();
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			if (!args.PropertyName.Equals(BackgroundAspectEffect.AspectProperty.PropertyName))
+				return;
+
+			cancellationTokenSource.Cancel();
```

Comment:
after this line, shouldn't we create a new instance?

Reply:
I've pushed a new commit. Interestingly when I cancel the token, dispose of it and new up a fresh instance it causes a wierd scenario where `Element` becomes null after going past the `Element == null` check and throwing a `NullReferenceException`. I thought once you cancelled a token you should dispose of it and get a new one, but it seems to work better without doing this... 🤨